metrics/web/docs/features/public_dashboard/01_public_dashboard.md

```diff
@@ -137,3 +158,187 @@ Once we've chosen the implementation approach, let's consider the components dia
+
+### Program
+> Detailed solution description to class/method level.
+ 
+The implementation of the public dashboard feature involves the changes in the following modules:
+
+- [the `feature config` module;](#featureconfig-module)
+- [the `auth` module;](#auth-module)
+- [the `navigation` module;](#navigation-module)
+
+Consider the following subsections that describe each module changes in more detail.
+
+#### FeatureConfig module
+
+The first important part is adding the ability to track the feature state, so that we can change the behavior of our application correspondingly.
+
+To implement this, we should follow the next steps:
+- add the `isPublicDashboardFeatureEnabled` boolean field to the `FeatureConfig` model;
+- change the `FeatureConfigData`'s `fromJson()` and `toJson()` methods, so they include handling the new added field;
+- create the [`PublicDashboardFeatureConfigViewModel`](#PublicDashboardFeatureConfigViewModel) and initialize it in the `FeatureConfigNotifier` as the other ones.
+
+Let's review the following class diagram, which describes the relationship between the classes in this module:
+
+![Feature Config Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_feature_config_class_diagram.puml)
+
+#### Auth module
+
+The next one, we should add the ability to sign-in anonymously. Also, we should know whether the user sign-in anonymously. For these purposes, we should make the following changes in the `auth` module:
+- add the `signInAnonymously()` method to the `UserRepository` abstract class;
+- add the implementation of the `signInAnonymously()` method to the `FirebaseUserRepository` class;
+- create the [`SignInAnonymouslyUseCase`](#SignInAnonymouslyUseCase) class, which calls the `signInAnonymously()` method of the `UserRepository`;
+- create the [`UserProfileViewModel`](#UserProfileViewModel) class;
+- create the `AuthState` enum;
+- integrate the use case, the enum, and the view model described above into the `AuthNotifier`.
+
+Let's review the following class diagram, which describes the relationship between the classes in this module:
+
+![Auth Class Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/public_dashboard_design/metrics/web/docs/features/public_dashboard/diagrams/public_dashboard_auth_class_diagram.puml)
+
+#### Navigation module
+
+The last thing, we should have the ability to manage the navigation of the anonymous user, that is, to know which pages they can visit and which they can't. Following the next steps, helps to achieve this purpose:
+- add the field of the `AuthState` type to the `NavigationNotifier`;
+- change the `handleAuthenticationUpdates()` method of the `NavigationNotifier` to take an argument of the type `AuthState`;
```

Comment:
Take into account that according to this [document](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/features/deep_links/01_deep_linking.md#deep-linking-with-authorization), the "handleAuthenticationUpdates()" method will change, so I think we should use updated information.